Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/NodePaletteWidget.h

```diff
@@ -56,22 +60,18 @@ namespace EMStudio
 
         void Init(EMotionFX::AnimGraph* animGraph, EMotionFX::AnimGraphNode* node);
 
-        static QIcon GetNodeIcon(const EMotionFX::AnimGraphNode* node);
-
     private slots:
         void OnFocusChanged(const QModelIndex& newFocusIndex, const QModelIndex& newFocusParent, const QModelIndex& oldFocusIndex, const QModelIndex& oldFocusParent);
 
     private:
-        void SaveExpandStates();
-        void RestoreExpandStates();
 
         AnimGraphPlugin*            m_plugin;
-        NodePaletteModel*           m_model;
-        QTreeView*                  m_treeView;
+        AZStd::unique_ptr <NodePaletteModelHelper> m_modelHelper;
```

Comment:
update: As the type this is being used in is a QWidget and we're not always sure about the lifetime, I'd have this use a raw pointer and `new` rather than a `AZStd::unique_ptr` to see if that makes any difference with the memory issues you were running into.

Reply:
Now I realized that this doesn't have to be a pointer and can be a regular field. And this makes the question of memory management moot :)